projects/vdk-core/plugins/vdk-plugin-control-cli/src/vdk/plugin/control_cli_plugin/execution_skip.py

```diff
@@ -0,0 +1,164 @@
+
+        return job_execution_running
+
+    def _get_running_job_executions(
+        self, job_name, team_name
+    ) -> List[DataJobExecution]:
+        job_status = [
+            "submitted",
+            "running",
+        ]  # We query only for running or manual jobs
+        response = self.execution_api_client.data_job_execution_list(
+            job_name=job_name, team_name=team_name, execution_status=job_status
+        )
+        log.info(f"Api call returned job execution list: {response}")
+        return response
+
+    @staticmethod
+    def _authenticate(api_token_authorization_url, api_token, auth_type) -> None:
+        log.info("Authenticating.")
+        auth = Authentication()
+        auth.update_api_token_authorization_url(api_token_authorization_url)
+        auth.update_api_token(api_token)
+        auth.update_auth_type(auth_type)
+        auth.acquire_and_cache_access_token()
+        log.info("Login successful.")
+
+
+class ExitWithMessage:
+    def __init__(self) -> None:
+        super.__init__(self)
+
+    @staticmethod
+    def skip_job_run(job_name) -> None:
+        log.info(f"Data job: {job_name} is already running in K8S.")
+        log.info(
+            f"Skipping execution and exiting. You can retry the job when it finishes executing in K8S."
+        )
+        os._exit(0)
+
+
+def _skip_job_if_necessary(
+    log_config: str,
+    job_name: str,
+    execution_id: str,
+    job_team: str,
+    action: WriteToFileAction,
+):
+    try:
+        log.info("Checking if job should be skipped:")
+        log.info(
+            f"Job : {job_name}, Team : {job_team}, Log config: {log_config}, execution_id: {execution_id}"
+        )
+        vdk_config = VDKConfig()
+        # vdk_config.
+        if log_config != "CLOUD":
+            logging.info("Local execution, skipping parallel execution check.")
+            return None
+
+        rest_api_url = vdk_config.control_service_rest_api_url
+        api_token = vdk_config.api_token
+        auth_url = vdk_config.api_token_authorization_server_url
+        auth_type = "api-token"
+
+        job_checker = ConcurrentExecutionChecker(
+            rest_api_url, api_token, auth_url, auth_type
+        )
+        job_running = job_checker.is_job_execution_running(
+            job_name, job_team, execution_id
+        )
+
+        if job_running:
+            log.info(f"Skipping job {job_name}")
+            action.skipped()
+            ExitWithMessage.skip_job_run(job_name)  # calls os._exit(0)
+            return 1  # All other branches return None
+    except Exception as exc:
+        log.warning(
+            f"Error while checking for another data job excecution: {str(exc)} "
+        )
+        log.warning("Printing stack trace and proceeding with execution:")
+        log.warning(exc)
+    return None
+
+
+@hookimpl(tryfirst=True)
+def run_job(context: JobContext) -> None:
+    log.info("Plugin hook method. Obtaining arguments for skip execution check.")
```

Comment:
Maybe drop this log since it would be called on every job run and clutter the logs a bit too much.